### PR TITLE
Revises 'JitTimer::PrintCsvHeader' to use 'append' mode

### LINKED ISSUE #7072: Core CLR crashes when COMPlus_JitTimeLogCsv is set
When COMPlus_JitImeLogCsv is set, Core CLR crashes with the following message (in Ubuntu 14.04 x64):

```
$ COMPlus_JitTimeLogCsv="hello.csv" Linux.x64.Relase/corerun hello.dll
{0x7636-0x7fcfa9801ea0} ASSERT [CRT    ] at /home/parjong/projects/toolchains/coreclr/src/pal/src/cruntime/file.cpp.680: Expression: f != NULL
Aborted (core dumped)
```

### PR DESCRIPTION
**Pull request by parjong**
The current implementation of 'JitTimer::PrintCsvHeader' opens a CSV
file with 'read' mode and re-opens it with 'write' mode if it is absent.

The current implementation has two problems:
First, "read" file pointer will be closed even if it is null,
which is reported in #7072.
Second, "write" file pointer will be leaked as there is no corresponding
fclose.

This commit rewrites 'JitTimer::PrintCsvHeader' to fix #7072.

### CODE REVIEW

**Comment by erozenfeld** [comment]
@dotnet/jit-contrib

**Comment by BruceForstall** [comment]
Hmmm... I'm confused because https://github.com/dotnet/coreclr/pull/7067 converts it to be this buggy way, which I commented on, but apparently that buggy way was already merged? @pkukol what's going on here?

File: src/jit/compiler.cpp [hunk 1]
```diff
@@ -7202,32 +7202,30 @@ void JitTimer::PrintCsvHeader()
 
     CritSecHolder csvLock(s_csvLock);
 
-    FILE* fp = _wfopen(jitTimeLogCsv, W("r"));
-    if (fp == nullptr)
+    if (FILE* fp = _wfopen(jitTimeLogCsv, W("a")))
```
Comment (BruceForstall) [inline src/jit/compiler.cpp]: This is a bit too fancy, and doesn't follow the convention of explicitly comparing non-bools in an 'if'. Why not just:
```
FILE* fp = _wfopen(jitTimeLogCsv, W("a"));
if (fp != nullptr)
{
```

**Comment by parjong** [comment]
@BruceForstall The code of interest comes from #7028 (not #7067).

File: src/jit/compiler.cpp (hunk 1 above)
Comment (parjong) [inline src/jit/compiler.cpp]: I just want to restict the scope of ''fp'' in the if block. I'll revise the code style as you suggested.

**Comment by BruceForstall** [comment]
LGTM

**Comment by pkukol** [comment]
So should I undo my change, then? My first commit was missing a tiny edit (I did it in notepad and goofed), but I've corrected that. The comments make it clear that the intended behavior is to rewrite any existing files and my fix preserves that behavior, but this change doesn't so I'm confused now. Sorry if I'm just being dense.

**Comment by jkotas** [comment]
@parjong Is there still anything left that needs fixing? If there is, could you please resolve the conflict?

**Comment by parjong** [comment]
@jkotas It seems that the recent code still has the same issue (reported in #7072). I rebased and pushed PR. Please take a look.

\CC @dotnet/jit-contrib

**Comment by danmosemsft** [comment]
@jkotas could you take a final look ?

**Comment by jkotas** [comment]
This should be signed off by @pkukol or somebody else on @dotnet/jit-contrib

**Comment by danmosemsft** [comment]
@dotnet/jit-contrib @pkukol  ?

**Comment by danmosemsft** [comment]
ping @dotnet/jit-contrib @pkukol

**Pull request merged**